test/Tippy.test.js

```diff
@@ -53,6 +53,29 @@ describe('<Tippy />', () => {
     expect(tip.popper.querySelector('strong')).not.toBeNull()
   })
 
+  test('custom class name get added to DOM', () => {
+    const className = 'hello'
+    const { container } = render(
+      <Tippy content="tip content" className={className}>
```

Comment:
Can you add a test that checks the add/remove when updating the state? e.g. starts with className A, after updating, ends with className B, but doesn't still contain className A. the useEffect looks correct to me anyway but still :-)

Reply:
Also, I think this should actually be `useLayoutEffect`. Reason being, mutations to the DOM (tippy element in this case) need to be synchronous with every render, and not wait after paint. It might cause undesirable transitions or something?

Reply:
I'm afraid I may need a hand here. Two things, actually:

- from what I've read, `useLayoutEffect` does seem to be the right method to choose for this, but it throws a warning when running the test: "Warning: useLayoutEffect does nothing on the server, because its effect cannot be encoded..." (see full message: https://github.com/facebook/react/pull/14596/files). I'm not sure how to suppress that. Or possibly it's just not working properly when run in the test.
- secondly, I couldn't get the test to work as expected (Possibly it's actually a bug in the code?). Rerendering the component shows the old class didn't get removed - but testing manually in the browser seems okay.

Reply:
To be honest, I don't really know either. There seems to be missing documentation for this or maybe I can't find it??? 😢

Reply:
useLayoutEffect runs synchronously after each render, so it's good for things like DOM manipulation outside React (like plain DOM api, or jQuery). I'm gonna take a look at this PR today

Reply:
I changed it back to `useEffect` and it seems like `rerender()` doesn't invoke `useEffect` cleanup function. However, [React in browser environment does](https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects), so that's why it is working when testing manually.

I have no idea how to fix it either.

Reply:
I will go ahead and merge this since it's working fine in the browser and logic looks correct 🤷‍♂️ If we can figure out how to test it, make a PR